tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -535,15 +663,26 @@ def validate_executor_placements(cls, executor_placements):
           'Expected 3 elements in the `federated_reduce()` argument tuple, '
           'found {}.'.format(len(arg.internal_representation)))
 
-    val_type = arg.type_signature[0]
+    # Encrypt client tensors
+    pk_a, sk_a = await self._trusted_aggregator_generate_keys()
+    enc_clients_arg = await self._encrypt_client_tensors(arg, pk_a)
+    val_type = enc_clients_arg.type_signature
+    val = enc_clients_arg.internal_representation
+    # Store original client dtype before encryption for 
+    # future decryption
+    orig_clients_dtype = arg.type_signature[0].member.dtype
+
     py_typecheck.check_type(val_type, computation_types.FederatedType)
     item_type = val_type.member
     zero_type = arg.type_signature[1]
     op_type = arg.type_signature[2]
-    type_utils.check_equivalent_types(
-        op_type, type_factory.reduction_op(zero_type, item_type))
 
-    val = arg.internal_representation[0]
+    # Note: this type check is not working because of the new types
```

Comment:
we should be able to check this after decryption, no?

Reply:
oups, yes, fixed!